pkg/console/controllers/route/controller.go

```diff
@@ -116,8 +124,26 @@ func (c *RouteSyncController) Sync(ctx context.Context, controllerContext factor
 
 	statusHandler := status.NewStatusHandler(c.operatorClient)
 
+	infrastructureConfig, err := c.infrastructureClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("infrastructure config error: %v", err)
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	clusterVersionConfig, err := c.clusterVersionLister.Get("version")
+	if err != nil {
+		klog.Errorf("cluster version config error: %v", err)
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	// Disable the route check for external control plane topology (hypershift) if the ingress capability is disabled.
+	if util.IsExternalControlPlaneWithIngressDisabled(infrastructureConfig, clusterVersionConfig) {
```

Comment:
by doing this you are basically disabling the operator from doing its work by syncing the console routes.

Reply:
@jhadvig: From what I see both of the functions used by the route controller (`SyncCustomRoute`,`SyncDefaultRoute`) are related to the component route implementation (custom hostname or tls). The component routes are not supported on HyperShift, so I think we are safe to skip them.

@csrwng: can you please confirm that the component routes are not supported on HyperShift?

Reply:
That is correct. If you've disabled ingress you are also indirectly disabling component routes, no?

Reply:
Right, I put my followup comment into a wrong conversation. Here it is:

Also, in the absence of the ingress operator the required RBAC won't be created for the components (including the console operator): [Component routes chapter of the EP](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/optional-ingress-hypershift.md#component-routes).